console/v1/types_console_sample.go

```diff
@@ -0,0 +1,183 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleSample is an extension to customizing OpenShift web console by adding samples.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type ConsoleSample struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata"`
+
+	// +required
+	Spec ConsoleSampleSpec `json:"spec"`
+}
+
+// ConsoleSampleSpec is the desired sample for the web console.
+// Samples will appear with their title, descriptions and a badge in a samples catalog.
+type ConsoleSampleSpec struct {
+	// title is the display name of the sample.
+	//
+	// It is required and must be no more than 50 characters in length.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	Title string `json:"title"`
+
+	// abstract is a short teaser the sample.
+	//
+	// It is required and must be no more than 100 characters in length.
+	//
+	// Currently, the abstract is shown on the sample card tile below the title and provider
+	// and is limited to three lines of content.
+	// +required
+	// +kubebuilder:validation:MaxLength=100
+	Abstract string `json:"abstract"`
+
+	// description of the sample.
+	//
+	// It is required and can have a maximum length of **4096** characters.
+	//
+	// It is a README.md-like content for additional information, links, pre-conditions, and other instructions.
+	// It will be rendered as Markdown so that it can contain line breaks, links, and other simple formatting.
```

Comment:
```suggestion
	// description is a long form explanation of the sample.
	//
	// It is required and can have a maximum length of **4096** characters.
	//
	// It is a README.md-like content for additional information, links, pre-conditions, and other instructions.
	// It will be rendered as Markdown so that it can contain line breaks, links, and other simple formatting.
```